### PR TITLE
Fix text edit widget editable's logic

### PR DESCRIPTION
**Pull request by nirvn**
@lindacamathias , can you test these APKs when our diligent fairy delivers?

### CODE REVIEW

File: src/qml/editorwidgets/TextEdit.qml [hunk 1]
```diff
@@ -8,7 +8,7 @@ import "."
 EditorWidgetBase {
   id: topItem
 
-  property bool isEditable: isEnabled && value && typeof(value) !== 'object'
+  property bool isEditable: isEnabled && typeof(value) !== 'object'
```
Comment (suricactus) [inline src/qml/editorwidgets/TextEdit.qml:11]: Beware that `typeof(null) === 'object'`

Comment (nirvn) [reply src/qml/editorwidgets/TextEdit.qml:11]: Ok, I'll need to go at it another way, something was germinating at this very moment

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 509146e3cd0dd401d2959806a79f2233b5d8a85f:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2008-c9df1c8c8056c10f010bae17391f4d03e2264593-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2008-c9df1c8c8056c10f010bae17391f4d03e2264593-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2008-c9df1c8c8056c10f010bae17391f4d03e2264593-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2008-c9df1c8c8056c10f010bae17391f4d03e2264593-x86.apk)

**Comment by nirvn** [comment]
@suricactus , how's the alternative approach?

File: src/qml/editorwidgets/TextEdit.qml [hunk 2]
```diff
@@ -25,7 +28,7 @@ EditorWidgetBase {
     color: 'gray'
     wrapMode: Text.Wrap
 
-    text: value == null ? '' : stringUtilities.insertLinks(value)
+    text: { console.log(); return value == null ? '' : stringUtilities.insertLinks(value); }
```
Comment (suricactus) [inline src/qml/editorwidgets/TextEdit.qml:31]: `console.log();`?

Comment (nirvn) [reply src/qml/editorwidgets/TextEdit.qml]: Gone

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for c2935e16fc7266aa0fe830937783f2690396b80b:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2008-3f701dce8f1b3b9b2560908d9a425ef3482948ee-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2008-3f701dce8f1b3b9b2560908d9a425ef3482948ee-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2008-3f701dce8f1b3b9b2560908d9a425ef3482948ee-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2008-3f701dce8f1b3b9b2560908d9a425ef3482948ee-x86.apk)

**Review by suricactus — APPROVED** [review]
Looks good to me

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for c4c61e849a22714d73b81a198cde3065c03738c1:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2008-c148e6d92585163dc76a67cb5c0f7a6c9a342314-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2008-c148e6d92585163dc76a67cb5c0f7a6c9a342314-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2008-c148e6d92585163dc76a67cb5c0f7a6c9a342314-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2008-c148e6d92585163dc76a67cb5c0f7a6c9a342314-x86.apk)

File: src/qml/editorwidgets/TextEdit.qml (hunk 1 above)
Comment (suricactus) [reply src/qml/editorwidgets/TextEdit.qml]: you can always do `value === null || typeof(value) !== 'object'`

Comment (nirvn) [reply src/qml/editorwidgets/TextEdit.qml]: Yeah but ultimately I think it makes more sense to rely on field type here.

**Pull request merged**